### PR TITLE
feat: Rename Fees to Revenue and add Event Date in Admin Sales Fees.

### PR DESCRIPTION
**Pull request by Anupam-dagar**
Reference: #3173 

#### Short description of what this resolves:
Currently there is no event date in the table of fees in admin sales section. This PR adds an event date to the table of fees in admin sales section. Also the Fees is renamed to Revenue.

#### Changes proposed in this pull request:

- Rename Fees to Revenue.
- Fetch events model along with admin-sales-fee model.
- Sort both models by name so that while adding event date for an event in admin-sales-fee, mismatch of event isn't there.
- Add aftermodel to add event date to admin-sales-fee model.
- Make relevant changes in the frontend.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Screenshots
Screenshot from 2019-06-28 01-41-57

### CODE REVIEW

File: app/routes/admin/sales/fees.js [hunk 1]
```diff
@@ -1,11 +1,23 @@
 import Route from '@ember/routing/route';
+import RSVP from 'rsvp';
 
 export default Route.extend({
   titleToken() {
-    return this.l10n.t('Fees');
+    return this.l10n.t('Revenue');
```
Comment (niranjan94) [inline app/routes/admin/sales/fees.js]: The route name also should chan ge

File: app/routes/admin/sales/fees.js [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
   },
 
   model() {
-    return this.store.findAll('admin-sales-fee');
+    return RSVP.hash({
```
Comment (niranjan94) [inline app/routes/admin/sales/fees.js]: Use async-await instead of RSVP.hash

File: app/routes/admin/sales/fees.js [hunk 3]
```diff
@@ -1,11 +1,23 @@
...
+      orders : this.store.findAll('admin-sales-fee', { reload: true }).then(orders => orders.sortBy('name')),
+      events : this.store.findAll('event', { reload: true }).then(events => events.sortBy('name'))
+    });
+  },
+
+  afterModel(model) {
+    model.orders.forEach((data, index) => data.set('eventDate', model.events[index].startsAtDate));
```
Comment (niranjan94) [inline app/routes/admin/sales/fees.js]: Do this in model

File: app/routes/admin/sales/fees.js [hunk 4]
```diff
@@ -1,11 +1,23 @@
...
+  },
+
+  setupController(controller, model) {
```
Comment (niranjan94) [inline app/routes/admin/sales/fees.js]: Do not do this.

**Comment by Anupam-dagar** [comment]
@niranjan94 This is ready for another review.

File: app/routes/admin/sales/revenue.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+export default Route.extend({
+  titleToken() {
+    return this.l10n.t('Revenue');
+  },
+
+  async model() {
+    let orders = await this.store.findAll('admin-sales-fee', { reload: true }).then(orders => orders.sortBy('name'));
+    let events = await this.store.findAll('event', { reload: true }).then(events => events.sortBy('name'));
+    orders.forEach((data, index) => data.set('eventDate', events[index].startsAtDate));
```
Comment (niranjan94) [inline app/routes/admin/sales/revenue.js]: Please do not assume the indexes of both orders and events match. (ie) for example, do not assume `orders[0]` matches with `events[0]`

Comment (Anupam-dagar) [reply app/routes/admin/sales/revenue.js]: I actually assumed it on the basis that I sorted them on the basis of name. I guess name in orders and events are event name only, right? I wasn't able to find another way to get event date as their is no relation to event in orders model. Maybe we can discuss this in tomorrow's meeting.

**Comment by Anupam-dagar** [comment]
This PR has now a dependency on corresponding server PR https://github.com/fossasia/open-event-server/pull/6135

File: app/routes/admin/sales/revenue.js [hunk 6]
```diff
@@ -2,10 +2,10 @@ import Route from '@ember/routing/route';
 
 export default Route.extend({
   titleToken() {
-    return this.l10n.t('Fees');
+    return this.l10n.t('Revenue');
   },
 
-  model() {
+  async model() {
     return this.store.findAll('admin-sales-fee');
```
Comment (CosmicCoder96) [inline app/routes/admin/sales/revenue.js]: not required here, if you are not using await.

File: app/routes/admin/sales/revenue.js [hunk 7]
```diff
@@ -2,10 +2,10 @@ import Route from '@ember/routing/route';
 
 export default Route.extend({
   titleToken() {
-    return this.l10n.t('Fees');
+    return this.l10n.t('Revenue');
   },
 
-  model() {
+  async model() {
```
Comment (niranjan94) [inline app/routes/admin/sales/revenue.js]: async not needed

File: app/models/admin-sales-fee.js [hunk 8]
```diff
@@ -1,11 +1,13 @@
 import attr from 'ember-data/attr';
 import ModelBase from 'open-event-frontend/models/base';
+import moment from 'moment';
```
Comment (shreyanshdwivedi) [inline app/models/admin-sales-fee.js]: Never used. Please remove this as travis is failing due to this

**Comment by Anupam-dagar** [comment]
This PR is ready for another review.

**Comment by Anupam-dagar** [comment]
@CosmicCoder96 yes the server PR is merged.

**Comment by Anupam-dagar** [comment]
can we merge this?

**Pull request merged**